tests/test_functional.py

```diff
@@ -0,0 +1,397 @@
+        self._linter.set_reporter(test_reporter)
+        self._linter.config.persistent = 0
+        checkers.initialize(self._linter)
+        self._linter.disable('I')
+        try:
+            self._linter.read_config_file(test_file.option_file)
+            self._linter.load_config_file()
+        except NoFileError:
+            pass
+
+        self._test_file = test_file
+
+    def setUp(self):
+        if self._should_be_skipped_due_to_version():
+            pytest.skip(
+                'Test cannot run with Python %s.' %
+                (sys.version.split(' ')[0],))
+        missing = []
+        for req in self._test_file.options['requires']:
+            try:
+                __import__(req)
+            except ImportError:
+                missing.append(req)
+        if missing:
+            pytest.skip('Requires %s to be present.' % (','.join(missing),))
+        if self._test_file.options['except_implementations']:
+            implementations = [
+                item.strip() for item in
+                self._test_file.options['except_implementations'].split(",")
+            ]
+            implementation = platform.python_implementation()
+            if implementation in implementations:
+                pytest.skip(
+                    'Test cannot run with Python implementation %r'
+                    % (implementation, ))
+
+    def _should_be_skipped_due_to_version(self):
+        return (sys.version_info < self._test_file.options['min_pyver'] or
+                sys.version_info > self._test_file.options['max_pyver'])
+
+    def __str__(self):
+        return "%s (%s.%s)" % (self._test_file.base, self.__class__.__module__,
+                               self.__class__.__name__)
+
+    def _open_expected_file(self):
+        return open(self._test_file.expected_output)
+
+    def _open_source_file(self):
+        if self._test_file.base == "invalid_encoded_data":
+            return open(self._test_file.source)
+        else:
+            return io.open(self._test_file.source, encoding="utf8")
+
+    def _get_expected(self):
+        with self._open_source_file() as fobj:
+            expected_msgs = get_expected_messages(fobj)
+
+        if expected_msgs:
+            with self._open_expected_file() as fobj:
+                expected_output_lines = parse_expected_output(fobj)
+        else:
+            expected_output_lines = []
+        return expected_msgs, expected_output_lines
+
+    def _get_received(self):
+        messages = self._linter.reporter.messages
+        messages.sort(key=lambda m: (m.line, m.symbol, m.msg))
+        received_msgs = collections.Counter()
+        received_output_lines = []
+        for msg in messages:
+            received_msgs[msg.line, msg.symbol] += 1
+            received_output_lines.append(OutputLine.from_msg(msg))
+        return received_msgs, received_output_lines
+
+    def _runTest(self):
+        self._linter.check([self._test_file.module])
```

Comment:
You could temporarily modify `sys.path` before and after this to fix your import-related issues.

Right now it's looking in all of the dirs in `sys.path` for `tests.shopify_python.functional`, but you could modify `sys.path` to include the contents of `tests/shopify_python` (first in the list of paths to search) and then change the `module` function to return `functional.cond_expr`, so that it would search for `functional/cond_expr.py` and find it within `tests/shopify_python` first.